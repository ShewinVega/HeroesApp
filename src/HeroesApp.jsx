import { Outlet } from 'react-router';

import {Navbar} from "./ui/index";

function HeroesApp() {
  return (
    <>
        <Navbar />
        <div className="container">
            <Outlet />
        </div>
    </>
  )
}

export default HeroesApp
