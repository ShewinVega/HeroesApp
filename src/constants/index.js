export default {
  MARVEL_PUBLISHER: 'Marvel Comics',
  DC_PUBLISHER: 'DC Comics',
}