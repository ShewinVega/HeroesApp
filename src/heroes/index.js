export * from './pages/MarvelPage';
export * from './pages/DcPage';
export * from './pages/HeroPage';
export * from './pages/SearchPage';