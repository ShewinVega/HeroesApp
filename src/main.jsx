import { createRoot } from 'react-dom/client';
import { BrowserRouter } from 'react-router';

import { AppRouter } from './router/AppRouter';
import { AuthProvider } from './context/Auth';

createRoot(document.getElementById('root')).render(
    <AuthProvider>
        <BrowserRouter>
            <AppRouter />
        </BrowserRouter>
    </AuthProvider>,
)
