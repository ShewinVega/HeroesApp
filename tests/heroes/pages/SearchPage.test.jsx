import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { SearchPage } from '../../../src/heroes';

const mockUseNavigate = jest.fn();
jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useNavigate: () => mockUseNavigate,
}));

describe(`Testing Search page`, () => {


  beforeEach(() => jest.clearAllMocks());

  test(`Should show correctly with default values`, () => {

    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test(`Should Show Batman and the input value with the queryString`, () => {

    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');
    // expect(input.value).toBe('batman');

    const img = screen.getByRole('img');
    console.log(img.src);
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg');

    const alertDanger = screen.getByLabelText('alert-primary');
    expect(alertDanger.style.display).toBe('none');

    // screen.debug();

  });

  test(`Should show an error if the hero does not exist. example(batman007)`, () => {

    render(
      <MemoryRouter initialEntries={['/search?q=batman11']}>
        <SearchPage />
      </MemoryRouter>
    );

    // screen.debug();
    const alertDanger = screen.getByLabelText('alert-danger');
    expect(alertDanger.style.display).not.toBe('none');
  });

  test(`Should call the navigate to a the new page`, () => {
    const query = `superman`;

    render(
      <MemoryRouter initialEntries={[`/search`]}>
        <SearchPage />
      </MemoryRouter>
    );


    const getInput = screen.getByRole('textbox');
    fireEvent.change(getInput, { target: { name: 'searchText' ,value: query } });

    const getForm = screen.getByRole('form');
    fireEvent.submit(getForm);

    expect(mockUseNavigate).toHaveBeenCalledWith(`?q=${query}`);
  });

});
