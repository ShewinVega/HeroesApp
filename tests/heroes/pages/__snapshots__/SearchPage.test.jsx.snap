// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Search page Should show correctly with default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        role="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="alert-primary"
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search Hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        There is no results with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
